controllers/storagecluster/exporter.go

```diff
@@ -215,3 +221,139 @@ func CreateOrUpdateServiceMonitor(r *StorageClusterReconciler, instance *ocsv1.S
+			}},
+			Labels: map[string]string{
+				"app.kubernetes.io/component": metricExporterName,
+				"app.kubernetes.io/name":      metricExporterName,
+				"app.kubernetes.io/version":   "0.0.1",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			ProgressDeadlineSeconds: &progressDeadlineSeconds,
+			Replicas:                &replicas,
+			RevisionHistoryLimit:    &revisionHistoryLimit,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/component": metricExporterName,
+					"app.kubernetes.io/name":      metricExporterName,
+				},
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxSurge:       &maxSurge,
+					MaxUnavailable: &maxUnavailable,
+				},
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app.kubernetes.io/component": metricExporterName,
+						"app.kubernetes.io/name":      metricExporterName,
+						"app.kubernetes.io/version":   "0.0.1"},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Args:    []string{"--namespaces", instance.Namespace},
+						Command: []string{"/usr/local/bin/metrics-exporter"},
+						Env: []corev1.EnvVar{
+							{
+								Name:  "WATCH_NAMESPACE",
+								Value: instance.Namespace,
+							},
+						},
+						Image: r.images.OCSMetricsExporter,
+						Name:  metricExporterName,
+						Ports: []corev1.ContainerPort{
+							{ContainerPort: 8080},
+							{ContainerPort: 8081},
+						},
+						SecurityContext: &corev1.SecurityContext{
+							RunAsNonRoot: truePtr,
+						},
+						VolumeMounts: []corev1.VolumeMount{{
+							Name:      "ceph-config",
+							MountPath: "/etc/ceph",
+						}},
+					}},
+					ServiceAccountName: metricExporterName,
+					Volumes: []corev1.Volume{{
+						Name: "ceph-config",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "ocs-metrics-exporter-ceph-conf",
+								},
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+	return dep
+}
+
+func deployMetricExporter(ctx context.Context, r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	expectedDeployment := getMetricExporterDeployment(r, instance)
+	namespacedName := types.NamespacedName{
+		Name:      expectedDeployment.Name,
+		Namespace: expectedDeployment.Namespace,
+	}
+	r.Log.Info("Reconciling metrics exporter service", "NamespacedName", namespacedName)
+
+	currentDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      expectedDeployment.Name,
+			Namespace: expectedDeployment.Namespace,
+		},
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, currentDeployment, func() error {
+		affirmMetricExporterDeployment(currentDeployment, expectedDeployment)
```

Comment:
Let's just inline this. No need for extra hops.

Reply:
Done